host/tests/hdmiCEC_L3_Tests/hdmiCEC_test01_TransmitCECCommands.py

```diff
@@ -0,0 +1,140 @@
+    """
+    A class for testing HDMI-CEC functionality by transmitting and verifying CEC commands.
+    """
+    def __init__(self, log:logModule=None):
+        """
+        Initializes the test class with test-specific configurations.
+
+        Args:
+            log (logModule): Logging module instance to record test execution.
+        """
+        # Class variables
+        self.testName  = "test01_TransmitCECCommands"
+        self.qcID = '1'
+        self.tvLogicalAddress = '0'
+        self.broadcastAddress = 'f'
+
+        # Initialize the parent class
+        super().__init__(self.testName, self.qcID, log)
+
+    def testFunction(self):
+        """
+        Main test function for transmitting HDMI-CEC commands and validating responses.
+
+        Steps:
+        1. Initialize HDMI-CEC module.
+        2. Add a logical address to the test device.
+        3. Get the logical address of the device.
+        4. List all connected CEC devices.
+        5. Iterate through devices and send commands to appropriate logical addresses.
+        6. Validate transmission and record results.
+        7. Clean up by removing logical addresses and terminating the HDMI-CEC module.
+
+        Returns:
+            bool: Final result of the test execution.
+        """
+
+        # Initialize the hdmiCEC module
+        self.testhdmiCEC.initialise()
+
+        # Add the logical Address.
+        self.testhdmiCEC.addLogicalAddress(self.tvLogicalAddress)
+
+        # Get the logical Address.
+        deviceLogicalAddress = self.testhdmiCEC.getLogicalAddress()
+
+        # List all connected CEC devices
+        self.cecDevices = self.hdmiCECController.listDevices()
+
+        # Final test result
+        finalResult = True
+
+        self.hdmiCECController.startMonitoring()
+
+        for device in self.cecDevices:
+            logicalAddress = device["logical address"]
+
+            # Skip sending messages to TV
+            if logicalAddress == '0' or logicalAddress == 'e':
+                continue
+
+            for command in self.cecCommands:
+                result = False
+                cec = command.get("command")
+                payload = command.get("payload")
+                type = command.get("type")
+
+                # Determine the destination logical address
+                destinationLogicalAddress = logicalAddress
+                if type == "Broadcast":
+                    destinationLogicalAddress = self.broadcastAddress
+
+                # Transmit the CEC command
+                self.testhdmiCEC.cecTransmitCmd(destinationLogicalAddress, cec, payload)
+
+                time.sleep(2)
+
+                self.log.stepStart(f'HdmiCecTx Source: {deviceLogicalAddress} Destination: {destinationLogicalAddress} CEC OPCode: {cec} Payload: {payload}')
+
+                # Validate the transmission
+                result = self.hdmiCECController.checkTransmitStatus(deviceLogicalAddress, destinationLogicalAddress, cec, payload)
```

Comment:
We need to validate two parameters here.  Staus and the result.  Also it differs between Tx and Broadcast.

Reply:
checkTransmitStatus reads the data bytes from the cec adaptor and validates, not from the device